src/App.tsx

```diff
@@ -1,14 +1,61 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { FilterType } from './types/FilterTypes';
+
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+const getVisibleTodos = (
+  todos: Todo[],
+  filterType: string,
+  query: string,
+): Todo[] => {
+  let visibleTodos: Todo[] = [...todos];
+
+  switch (filterType) {
+    case FilterType.ACTIVE:
+      visibleTodos = visibleTodos.filter(todo => !todo.completed);
+      break;
+
+    case FilterType.COMPLETED:
+      visibleTodos = visibleTodos.filter(todo => todo.completed);
+      break;
+
+    default:
+      break;
+  }
+
+  return visibleTodos.filter(
+    todo => todo.title.toLowerCase().includes(
+      query.toLowerCase(),
+    ),
+  );
+};
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedType, setSelectedType] = useState(FilterType.ALL);
+  const [hasError, setHasError] = useState(false);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setHasError(true))
+      .finally(() => setIsLoading(true));
```

Comment:
The name of the variable `isLoading` doesn't meet the expectations of what you store there. Before calling `getTodos` you need to call `setIsLoading(true)` because request start loading and in finally block you need to call `setIsLoading(false)`, because requests was finished. 
Because of this you used a really confusing condition below. You show loader when it's loading...

```
 {isLoading
  ? (
    <TodoList
      todos={visibleTodos}
      selectedModal={selectedTodo}
      onSelectedModal={setSelectedTodo}
    />
  )
  : (
    <Loader />
  )}
```